x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_single_layer_source_settings.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import React, { Fragment, Component, ChangeEvent } from 'react';
+import { EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
+import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
+import { MVTFieldConfigEditor } from './mvt_field_config_editor';
+import { MVTFieldDescriptor } from '../../../../common/descriptor_types';
+
+export type MVTSettings = {
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+};
+
+export interface State {
```

Comment:
Why are any of these tracked in state? `MVTFieldConfigEditor` tracks `currentFields` because local state allows for invalid state that should not be passed to redux. In this case, layerName, minSourceZoom, maxSourceZoom, and fields are just instantly passed to redux. Since you are not tracking anything different, just keep all state in redux and remove them from component state.

Reply:
discussed. This is just to keep the external interface simple. It's a trade-off vis-a-vis duplicating state. The component is used in the update-source-editor (layer settings) and the add-layer-wizard. The former has granular events for each source-property, the latter doesn't.